tests/sentry/models/test_dashboard.py

```diff
@@ -0,0 +1,60 @@
+from sentry.models.dashboard import Dashboard
+from sentry.testutils.cases import TestCase
+
+
+class IncrementalNameTest(TestCase):
```

Comment:
Since the original issue was to be able to successfully do a POST request with >10 dashboards, could we mock something similar in the endpoint tests to actually verify that? e.g. these tests have great coverage but it doesn't guarantee someone won't mess up how this is being used in the endpoint

Reply:
Added some more endpoint tests to cover this!